### PR TITLE
ONYX-15241: Add fixing line in conjur provider to support authn-jwt

### PR DESCRIPTION
**Pull request by tzheleznyak**
Allow authn-jwt flow in conjur provider 

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

ONYX-15241

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: internal/providers/conjur/provider.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func ProviderFactory(options plugin_v1.ProviderOptions) (plugin_v1.Provider, err
 		if conjur, err = conjurapi.NewClientFromTokenFile(provider.Config, tokenFile); err != nil {
 			return nil, fmt.Errorf("ERROR: Could not create new Conjur provider: %s", err)
 		}
-	case provider.AuthnURL != "" && strings.Contains(provider.AuthnURL, "authn-k8s"):
+	case provider.AuthnURL != "" && (strings.Contains(provider.AuthnURL, "authn-k8s") || strings.Contains(provider.AuthnURL, "authn-jwt")):
 		// Conjur provider using authenticator
 		log.Printf("Info: Conjur provider using Kubernetes authenticator-based authentication")
```
Comment (benoaviram) [inline internal/providers/conjur/provider.go:99]: in case of JWT this info isn't accurate..

**Comment by benoaviram** [comment]
lgtm

**Review by benoaviram — DISMISSED** [review]

File: internal/providers/conjur/provider.go [hunk 2]
```diff
@@ -94,9 +94,8 @@ func ProviderFactory(options plugin_v1.ProviderOptions) (plugin_v1.Provider, err
 		if conjur, err = conjurapi.NewClientFromTokenFile(provider.Config, tokenFile); err != nil {
 			return nil, fmt.Errorf("ERROR: Could not create new Conjur provider: %s", err)
 		}
-	case provider.AuthnURL != "" && strings.Contains(provider.AuthnURL, "authn-k8s"):
-		// Conjur provider using authenticator
-		log.Printf("Info: Conjur provider using Kubernetes authenticator-based authentication")
+	case provider.AuthnURL != "" && (strings.Contains(provider.AuthnURL, "authn-k8s") || strings.Contains(provider.AuthnURL, "authn-jwt")):
```
Comment (szh) [inline internal/providers/conjur/provider.go:97]: Can you use a slice (array) of valid authn types instead of calling contains() for each option?

Comment (szh) [reply internal/providers/conjur/provider.go:97]: Or else you could move this line to a helper function with a descriptive name?

**Review by szh — APPROVED** [review]
LGTM

**Pull request merged**